packages/backend-defaults/src/CreateBackend.test.ts

```diff
@@ -69,4 +76,91 @@ describe('createBackend', () => {
       }),
     ).toThrow('The core.pluginMetadata service cannot be overridden');
   });
+
+  it('should throw if an unsupported InternalSharedEnvironment version is passed in', () => {
+    expect(() =>
+      createBackend({
+        env: {} as any,
+      }),
+    ).toThrow(
+      "Shared environment version 'undefined' is invalid or not supported",
+    );
+    expect(() =>
+      createBackend({
+        env: { version: {} } as any,
+      }),
+    ).toThrow(
+      "Shared environment version '[object Object]' is invalid or not supported",
+    );
+    expect(() =>
+      createBackend({
+        env: { version: 'v2' } as any,
+      }),
+    ).toThrow("Shared environment version 'v2' is invalid or not supported");
+  });
+
+  it('should prioritize services correctly', async () => {
+    const backend = createBackend({
+      env: createSharedEnvironment({
+        services: [
+          createServiceFactory({
+            service: coreServices.rootHttpRouter,
+            deps: {},
+            async factory() {
+              return {
+                use() {},
+              };
+            },
+          }),
+          mockConfigFactory({
+            data: { root: 'root-env' },
+          }),
+          createServiceFactory({
+            service: fooServiceRef,
+            deps: {},
+            async factory() {
+              return 'foo-env';
+            },
+          }),
+          createServiceFactory({
+            service: barServiceRef,
+            deps: {},
+            async factory() {
+              return 'bar-env';
+            },
+          }),
+        ],
+      })(),
```

Comment:
So here we're not going to do the thing where createSharedEnvironment only returns a function if its input was a function? :)

Reply:
Nah, always a function right? Feeling it makes sense even for this somewhat more internal helper. Makes it easier to keep backwards compatibility when using this in an internally shared package too.